L1Trigger/L1TZDC/plugins/L1TZDCAnalyzer.cc

```diff
@@ -0,0 +1,154 @@
+
+  private:
+    void beginJob() override;
+    void analyze(const edm::Event&, const edm::EventSetup&) override;
+    void endJob() override;
+
+    //fileservice
+    edm::Service<TFileService> m_fs;
+    //Declare a tree, member and pointer
+    TTree* m_zdcEtSumTree_p;
+    //Declare the etSum max and bpx max
+    //MAX Here is effectively a hardcoded number - you just need to know it from the producer - realistically this is best handled by something passed but for now (2023.08.17) we hardcode
+    static const int m_maxBPX = 5;
+    float m_zdcEtSumP[m_maxBPX];
+    float m_zdcEtSumM[m_maxBPX];
+
+    //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+    //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+    //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+    //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+    // ----------member data ---------------------------
+    edm::EDGetToken m_sumToken;
+
+    bool doText_;
+    bool doHistos_;
+
+    TFileDirectory evtDispDir_;
+  };
+
+  //
+  // constants, enums and typedefs
+  //
+
+  //
+  // static data member definitions
+  //
+
+  //
+  // constructors and destructor
+  //
+  L1TZDCAnalyzer::L1TZDCAnalyzer(const edm::ParameterSet& iConfig)
+      : doText_(iConfig.getUntrackedParameter<bool>("doText", true)),
+        doHistos_(iConfig.getUntrackedParameter<bool>("doHistos", true)) {
+    usesResource(TFileService::kSharedResource);
+    //now do what ever initialization is needed
+
+    // register what you consume and keep token for later access:
+    edm::InputTag nullTag("None");
+
+    edm::InputTag sumTag = iConfig.getParameter<edm::InputTag>("etSumToken");
+    m_sumToken = consumes<l1t::EtSumBxCollection>(sumTag);
+
+    std::cout << "Processing " << sumTag.label() << std::endl;
+  }
+
+  L1TZDCAnalyzer::~L1TZDCAnalyzer() {
+    // do anything here that needs to be done at desctruction time
+    // (e.g. close files, deallocate resources etc.)
+  }
+
+  //
+  // member functions
+  //
+
+  // ------------ method called for each event  ------------
+  void L1TZDCAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+    using namespace edm;
+
+    std::stringstream text;
+
+    //    Handle<EtSumBxCollection> sums;
+    Handle<BXVector<l1t::EtSum> > sums;
+    iEvent.getByToken(m_sumToken, sums);
+
+    for (int ibx = sums->getFirstBX(); ibx <= sums->getLastBX(); ++ibx) {
+      for (auto itr = sums->begin(ibx); itr != sums->end(ibx); ++itr) {
+        if (itr->getType() == l1t::EtSum::EtSumType::kZDCP)
+          m_zdcEtSumP[ibx] = itr->hwPt();
+        if (itr->getType() == l1t::EtSum::EtSumType::kZDCM)
+          m_zdcEtSumM[ibx] = itr->hwPt();
+      }
+    }
+
+    m_zdcEtSumTree_p->Fill();
+
+    if (doText_)
+      edm::LogVerbatim("L1TCaloEvents") << text.str();
```

Comment:
I see `text` created up above, but I do not see anywhere between here and there where anything gets put into the stream. Am I missing something obvious? Is this just leftover debugging material?